tests/protocol/test_hostplugin.py

```diff
@@ -539,42 +539,48 @@ def test_validate_page_blobs(self):
                         exp_method, exp_url, exp_data)
 
     def test_validate_http_request_when_uploading_logs(self):
```

Comment:
Probably not a good test for the current changes but just putting this comment here incase I forget later, make sure to add a test to ensure that failures are handled gracefully by the agent (for example if the PUT call fails and you return a ProtocolError).

Reply:
+1

the code to handle the error is there already so we should include the test in this pr

Reply:
Added a test to verify the host plugin method raises on a failed request. Since this exception is propagated to the protocol, which propagates it to the log uploading thread (coming in the next PR), I will add another test then to make sure we are handling it gracefully.